presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1756,6 +1757,83 @@ private void testPartitionPerScanLimit(Session session, HiveStorageFormat storag
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testPartitionPerScanLimitSessionProperty()
+    {
+        TestingHiveStorageFormat storageFormat = new TestingHiveStorageFormat(getSession(), HiveStorageFormat.DWRF);
+        testWithStorageFormat(storageFormat, this::testPartitionPerScanLimitSessionProperty);
+    }
+
+    private void testPartitionPerScanLimitSessionProperty(Session session, HiveStorageFormat storageFormat)
+    {
+        Session defaultSession = getSession();
+
+        String tableName = "test_session_partition_per_scan_limit";
+        String partitionsTable = "\"" + tableName + "$partitions\"";
+
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE " + tableName + " " +
+                "(" +
+                "  foo VARCHAR," +
+                "  part BIGINT" +
+                ") " +
+                "WITH (" +
+                "format = '" + storageFormat + "', " +
+                "partitioned_by = ARRAY[ 'part' ]" +
+                ") ";
+
+        assertUpdate(defaultSession, createTable);
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, tableName);
+        assertEquals(tableMetadata.getMetadata().getProperties().get(STORAGE_FORMAT_PROPERTY), storageFormat);
+        assertEquals(tableMetadata.getMetadata().getProperties().get(PARTITIONED_BY_PROPERTY), ImmutableList.of("part"));
+
+        // insert 500 partitions
```

Comment:
Not sure of the test runtime, but it looks like we can test the same functionality using much smaller number of partitions. This will reduce the number of files and partitions created.

Reply:
Updated with smaller number of partitions